### ISSUE TITLE
How to enable row selection without using the checkbox?

### ISSUE THREAD

**Issue by cheahkhing** [adazzle/react-data-grid#238]
Hi, is there a way to simply select a row whenever user click on any column? instead of just the checkbox?

**Comment by jpdriver** [comment]
hey @cheahkhing -- there isn't right now, but I guess it wouldn't be too hard to do..

we have a function called `onSelect` that we call when you click on a cell if you pass `enableCellSelect = true` as a prop. we also have `handleRowSelect` which is what gets called when you click one of the checkboxes.

what you could do is pass a new prop to the grid (something like `selectRowOnCellClick`). if that was true, you could call `handleRowSelect` within the cell click, and if it was false you would default to the existing implementation.

it would mean some of the features stop working for you (*possibly* edit -- i'm not sure about that though, and things like drag-down / copy / paste which all require a cell to be selected..)

could you tell me a bit about your use case so I can fully understand why you want it?

**Comment by cheahkhing** [comment]
Yes, it is actually quite a simple scenario.
I am trying to setup a simple grid that contains a bunch of data.
And, user will be able to use "arrow up" / "arrow down" to highlight the entire row and press "enter" / "space" to select them.
I don't really need cell edit or cell focus. If possible, i would want to disable that as well. The reason is user don't need to mess with each cell data, they just need to highlight and select the row.
Today, the grid already shows a pretty nice "hover" color when mouse over it, but it doesn't do that for using keyboard.
Basically the whole idea is to make sure whatever user can do with mouse, they should be able to do with keyboard as well. In this case, "mouse hover" to "highlight", "mouse click" to "select" ==> becomes "up/down arrow" to "highlight", and "enter/space" to select.
Again, it should be entire row and not just a particular cell.

**Comment by danfernand** [comment]
+1

**Comment by bakesteve** [comment]
@cheahkhing that sounds fairly straightforward. We track the selected cell and have some styles (albeit subtle ones) as you navigate. Are you looking to highlight the whole row, or individual cells?
If its rows, you can do that fairly easily with a custom `RowRenderer` - will ping some examples up here with the state you need to check
Equally, feels like something that would be nice to have so will also take a quick look at adding some styles you can override. So by default its subtle, but easy to then change that to whatever you want in good old css.

**Comment by txm** [comment]
Hi.  I'm trying to get this working.  My use case is, user clicks a row (or even certain cells as some cells in the row are editable) and I display more info about the row to the user.

I have:

<ReactDataGrid enableCellSelect={true}

and:

onSelect: function(cell) { console.log('ok')

The only clue I have is 3 propType warnings:

bundle.js:2775 Warning: Failed propType: Required prop `value` was not specified in `Cell`. Check the render method of `Row`.
bundle.js:2775 Warning: Failed propType: Required prop `value` was not specified in `SimpleCellFormatter`. Check the render method of `Cell`.
bundle.js:2775 Warning: Failed propType: Required prop `value` was not specified in `EditorContainer`. Check the render method of `Cell`.

**Comment by vedran-maestral** [comment]
@jpdriver 
In the previous post, you have mentioned:
_we have a function called onSelect that we call when you click on a cell if you pass enableCellSelect = true as a prop. we also have handleRowSelect which is what gets called when you click one of the checkboxes._

Thank you for your reply. 
I cant get it to work the onSelect. I need a current row when the user clicks on it.

**Comment by soma83** [comment]
@vedran-maestral 
Did you find a solution for the onSelect when enableCellSelect={true}??? I'm facing the very same problem. I can't get the row when the user selects it. And the onRowUpdated is obiously called only when a cell is modified...

**Comment by GenoD** [comment]
@soma83 little late but have you tried using onRowClick? It gives you rowIdx and row. With that you might be able to loop through to de-select all rows and set selected to true on row clicked.

**Comment by baranaytas** [comment]
you can call onRowsSelected on onRowClick

onRowClick={(rowId,row)=>this.onRowsSelected([{row:row,rowIdx:rowId}])}

**Comment by baranaytas** [comment]
<ReactDataGrid ref="location_analysis_grid"
                onGridSort={this.handleGridSort}
                enableCellSelect={false}
                columns={this.state.columnList}
                rowGetter={this.rowGetter}
                rowsCount={this.getSize()}
                minHeight={250}
                onAddFilter={this.handleFilterChange}
                onClearFilters={this.onClearFilters}
                enableRowSelect='single'
                onRowClick={(rowId,row)=>this.onRowsSelected([{row:row,rowIdx:rowId}])}
                rowSelection={{
                    showCheckbox: true,
                    enableShiftSelect: true,
                    onRowsSelected: this.onRowsSelected,
                    onRowsDeselected: this.onRowsDeselected,
                    multiSelect: false,
                    selectBy: { keys: { rowKey:'Id', values: this.state.selectedNodes },isSelectedKey: 'isSelected'}
                }}
            />

    onRowsSelected(rows) {
        debugger;
        this.setState({
            selectedNodes: this.state.selectedNodes.concat(rows.map(r => r.row.Id))
        }, function() {
            this.state.emptyGridSelection = false;
            this.props.selectedNodesChanged(this.state.selectedNodes, this.state.emptyGridSelection);
        });
    }

**Comment by ssad9985** [comment]
For not using checkbox to select a row, give 'showCheckbox:false' in 
Based on the jpdriver comment, I have made the following changes 

<ReactDataGrid
                selectRowOnCellClick={true}
              rowSelection={{
                 showCheckbox: false, //to hide checkbox in each row
                        />

In react-data-grid.js file, add --> 	selectRowOnCellClick: React.PropTypes.bool, 	//to select row without checkbox

Also in the cell click function,

onCellClick: function onCellClick(cell) {
	    this.onSelect({ rowIdx: cell.rowIdx, idx: cell.idx });

	    if (this.props.onRowClick && typeof this.props.onRowClick === 'function') {
	      this.props.onRowClick(cell.rowIdx, this.props.rowGetter(cell.rowIdx));
	    }
			//to select row without checkbox
			if(this.props.selectRowOnCellClick){
				this.handleRowSelect(cell.rowIdx, cell.columnKey, cell.rowData);
			}
	  }

This worked for me.!
